entity_parser.go

```diff
@@ -365,20 +368,48 @@ func parseETLTag(tag string) (string, ETLState, error) {
 	return fullETLTag, etlState, nil
 }
 
+// parseTTLTag functions parses DOSA "ttl" tag
+func parseTTLTag(tag string) (string, time.Duration, error) {
+	fullTTLTag := ""
+	ttlTag := ""
+	matches := ttlPattern0.FindStringSubmatch(tag)
+	if len(matches) == 2 {
+		fullTTLTag = matches[0]
+		ttlTag = matches[1]
+	}
+
+	if len(matches) == 0 {
+		return "", NoTTL(), nil
+	}
+
+	// filter out "trailing comma"
+	ttlTag = strings.TrimRight(ttlTag, " ,")
+	ttl, err := time.ParseDuration(ttlTag)
+	if err != nil {
+		return "", NoTTL(), err
+	}
+
+	if ttl < 0 {
+		return fullTTLTag, NoTTL(), nil
+	}
+
+	return fullTTLTag, ttl, nil
+}
+
 // parseEntityTag function parses DOSA tag on the "Entity" field
-func parseEntityTag(structName, dosaAnnotation string) (string, ETLState, *PrimaryKey, error) {
+func parseEntityTag(structName, dosaAnnotation string) (string, time.Duration, ETLState, *PrimaryKey, error) {
 	tag := dosaAnnotation
 
 	// find the primaryKey
 	matchs := primaryKeyPattern0.FindStringSubmatch(tag)
 	if len(matchs) != primaryKeyPattern0.NumSubexp()+1 {
-		return "", EtlOff, nil, fmt.Errorf("dosa.Entity on object %s with an invalid dosa struct tag %q", structName, tag)
+		return "", time.Duration(0), EtlOff, nil, fmt.Errorf("dosa.Entity on object %s with an invalid dosa struct tag %q", structName, tag)
```

Comment:
nit: let's be consistent with the rest of the code below and instead use `NoTTL()` here.